include/ensmallen_bits/moead/moead_impl.hpp

```diff
@@ -0,0 +1,383 @@
+        - 0.5 + iterate;
+  }
+
+  // 1.3 F-value initialisation for the population.
+  arma::mat FValue(numObjectives, populationSize);
+  EvaluateObjectives(population, objectives, FValue);
+
+  // 1.4 Initialize the ideal point z.
+  arma::mat idealPoint(numObjectives, 1);
+  std::vector<MatType> iterateWrapper(1);
+  iterateWrapper[0].resize(iterate.n_rows, iterate.n_cols);
+  iterateWrapper[0]=iterate;
+  EvaluateObjectives(iterateWrapper, objectives, idealPoint);
+
+  terminate |= Callback::BeginOptimization(*this, objectives, iterate, callbacks...);
+
+  // 2 The main loop.
+  for(size_t g = 0; g < numGeneration; g++)
+  {
+    shuffle = std::get<0>(objectives).Shuffle(populationSize);
+    for (size_t i : shuffle)
+    {
+      terminate |= Callback::StepTaken(*this, objectives, iterate, callbacks...);
+
+      // 2.1 Randomly select two indices in weightNeighbourIndices(i) and use them
+      // to make a child.
+      size_t k = weightNeighbourIndices(i, arma::randi(arma::distr_param(0,  neighbourhoodSize-1))),
+             l = weightNeighbourIndices(i, arma::randi(arma::distr_param(0,  neighbourhoodSize-1)));
+      if(k == l)
+      {
+        if(k == neighbourhoodSize-1)
+          k--;
+        else
+          k++;
+      }
+      std::vector<MatType> candidate(1);
+      double determiner1 = arma::randu();
+      if(determiner1 < crossoverProb)
+      {
+        candidate[0].resize(iterate.n_rows, iterate.n_cols);
+        for (size_t idx = 0;idx < iterate.n_rows; idx++)
+        {
+          double determiner2 = arma::randu();
+          if (determiner2 < 0.5)
+            candidate[0][idx] = population[k][idx];
+          else
+            candidate[0][idx] = population[l][idx];
+          if(candidate[0][idx] < lowerBound(idx))
+            candidate[0][idx] = lowerBound(idx);
+          if(candidate[0][idx]>upperBound(idx))
+            candidate[0][idx] = upperBound(idx);
+        }
+      }
+      else
+        candidate[0] = population[i];
+
+      // 2.2 Improve the child.
+      Mutate(candidate[0], 1 / numObjectives, lowerBound, upperBound);
+
+      // Store solution for candidate.
+      arma::mat evaluatedCandidate(numObjectives, 1);
+      EvaluateObjectives(candidate, objectives, evaluatedCandidate);
+
+      // 2.3 Update of ideal point.
+      for (size_t idx = 0;idx < numObjectives;idx++)
+      {
+        idealPoint(idx, 0) = std::min(idealPoint(idx, 0),
+            evaluatedCandidate(idx, 0));
+      }
+
+      // 2.4 Update of the neighbouring solutions.
+      for (size_t idx = 0;idx < neighbourhoodSize;idx++)
+      {
+        if (DecomposedSingleObjective(weights.col(weightNeighbourIndices(i, idx)),
+              idealPoint.col(0), evaluatedCandidate.col(0))
+            <= DecomposedSingleObjective(
+              weights.col(weightNeighbourIndices(i,idx)),
+              idealPoint.col(0), FValue.col(weightNeighbourIndices(i, idx))))
```

Comment:
Minor style issue, the lines should align with the beginning of `DecomposedSingleObjective`.